packages/coreutils/src/ratelimiter.ts

```diff
@@ -108,9 +111,9 @@ export class Debouncer<T = any, U = any> extends RateLimiter<T, U> {
    * Invokes the function and only executes after rate limit has elapsed.
    * Each invocation resets the timer.
    */
-  async invoke(): Promise<T> {
+  async invoke(): Promise<T | undefined> {
     void this.poll.schedule({ interval: this.limit, phase: 'invoked' });
-    return this.payload.promise;
+    return this.payload!.promise;
```

Comment:
@afshin Is it safe to assume that the payload is non-null here?

Reply:
Yes it is. Also, at some very near date, this library should be removed from coreutils and will come from `@lumino/polling` anyway.